### PR TITLE
Add support for multiple parent and child viewtypes

### PR DESCRIPTION
**Pull request by matthew-reilly**

### CODE REVIEW

**Comment by plackemacher** [comment]
@weddingwire/android ready for review

**Comment by plackemacher** [comment]
Closed and reopened after enabling Travis-CI support.

File: criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeChildDetail.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.bignerdranch.android.criminalintent;
+
+import java.util.Date;
+
+public class CrimeChildDetail extends CrimeChild {
```
Comment (ujavid) [inline criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeChildDetail.java:5]: is this what you would call... a _smooth_ criminal? :trollface:

File: criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeLab.java [hunk 2]
```diff
@@ -20,11 +21,20 @@ public static CrimeLab get(Context context) {
...
+                crime.setTitle("Crime #" + i);
+                crime.setSolved(i % 2 == 0);
+                mCrimes.add(crime);
+            } else {
+                Felony crime = new Felony();
+                crime.setTitle("Felony #" + i);
+                crime.setSolved(i % 2 == 0);
+                crime.setDetail("Details");
+                mCrimes.add(crime);
+            }
```
Comment (rubsnick) [inline criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeLab.java:38]: I don't quite get what this class is doing with the random. is this generating ID's? Some comments or javadoc would be helpful.

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 3]
```diff
@@ -144,57 +196,57 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
...
-     * Bind data to the {@link CVH} here.
+     * Bind data to the {@link ChildViewHolder} here.
      *
      * @param childViewHolder The {@code CVH} to bind data to
      * @param position The index in the list at which to bind
      * @param childListItem The child list item which holds that data to be
      *                      bound to the {@code CVH}
      */
-    public abstract void onBindChildViewHolder(CVH childViewHolder, int position, Object childListItem);
+    public abstract void onBindChildViewHolder(ChildViewHolder childViewHolder, int position, Object childListItem);
```
Comment (rubsnick) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java:249]: I like that these are more explicit but there's a chance that the repo owner won't like it. If we do want to create a PR for them then it's best not to do these changes in the future for better chances of getting it merged.

File: criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeLab.java (hunk 2 above)
Comment (matthew-reilly) [inline criminalintentsample/src/main/java/com/bignerdranch/android/criminalintent/CrimeLab.java:38]: This is just for the sample. It adds some `Felony` objects randomly through the set. I'll add some docs.

**Comment by matthew-reilly** [comment]
@plackemacher Merge merge? I'd like to use a release from here in my vendor PR.

**Pull request merged**